fastestimator/estimator.py

```diff
@@ -340,7 +339,7 @@ def _configure_loader(self, loader: Union[DataLoader, tf.data.Dataset]) -> Union
         new_loader = loader
         if isinstance(new_loader, DataLoader) and isinstance(self.network, TFNetwork):
             add_batch = True
-            if hasattr(loader.dataset, "dataset") and isinstance(loader.dataset.dataset, BatchDataset):
+            if hasattr(loader.dataset, "dataset") and loader.dataset.dataset.fe_is_batched:
```

Comment:
the member variable should be _fe_is_batched to indicate private var

Reply:
doesn't make it less janky.    in model, we have model.fe_compiled

Reply:
slightly less janky. we should have done _fe_compiled for the model. We use ._ everywhere else that we are inserting random stuff into objects I think

Reply:
also, do we need a hasattr guard here?

Reply:
instance check it is